20_beta_diversity.Rmd

```diff
@@ -247,6 +247,151 @@ As a final note, `mia` provides functions for the evaluation of additional dissi
 * `calculateOverlap`, `runOverlap` ()
 * `calculateDPCoA`, `runDPCoA` (Double Principal Coordinate Analysis)
 
+Redundancy analysis is similar to PCA, however, it takes into account covariates. 
+It aims to maximize the variance in respect of covariates. The results shows how much
+each covariate affects.
+
+```{r microbiome_RDA1}
+# Load required packages
+if(!require("vegan")){
+    install.packages("vegan")
+    library("vegan")
+}
+if(!require("stringr")){
+    install.packages("stringr")
+    library("stringr")
+}
+if(!require("knitr")){
+    install.packages("knitr")
+    library("knitr")
+}
+# Load data
+data(enterotype)
+# Covariates that are being analyzed
+variable_names <- c("ClinicalStatus", "Gender", "Age")
+
+# Apply relative transform
+enterotype <- transformSamples(enterotype, method = "relabundance")
+
+# Get assay
+assay <- t(assay(enterotype, "relabundance"))
+# Get colData
+coldata <- colData(enterotype)
+
+# Create a formula
+formula <- as.formula(paste0("assay ~ ", str_c(variable_names, collapse = " + ")) )
+
+# # Perform RDA
+rda <- rda(formula, data = coldata, scale = TRUE, na.action = na.exclude)
+
+# Initialize list for p-values
+rda_info <- list()
```

Comment:
Also wondering if we could/should add or replace example with `dbrda` / `capscale`. These two do the same but one had some added capabilities in terms of interpretation.

>     Distance-based redundancy analysis (dbRDA) is an ordination method
>      similar to Redundancy Analysis (‘rda’), but it allows
>      non-Euclidean dissimilarity indices, such as Manhattan or
>      Bray-Curtis distance. Despite this non-Euclidean feature, the
>      analysis is strictly linear and metric. If called with Euclidean
>      distance, the results are identical to ‘rda’, but dbRDA will be
>      less efficient. Functions ‘capscale’ and ‘dbrda’ are constrained
>      versions of metric scaling, a.k.a. principal coordinates analysis,
>      which are based on the Euclidean distance but can be used, and are
>      more useful, with other dissimilarity measures. The functions can
>      also perform unconstrained principal coordinates analysis,
>      optionally using extended dissimilarities.
>

Reply:
This is rather important, and standard. I suggest to add before the first merge already.